cmd/neofs-node/tree.go

```diff
@@ -38,6 +40,16 @@ func initTreeService(c *cfg) {
 		c.treeService.Start(ctx)
 	}))
 
+	syncTreeFunc := func(ctx context.Context) {
+		syncTrees(ctx, c.treeService, c.shared.cnrClient, c.log)
+	}
+
+	if c.cfgNetmap.state.controlNetmapStatus() == control.NetmapStatus_ONLINE {
+		c.workers = append(c.workers, newWorkerFromFunc(syncTreeFunc))
+	} else {
+		c.addOnlineStateHandler(syncTreeFunc)
```

Comment:
What happens if we start ONLINE, go OFFLINE and then ONLINE again?

Reply:
good question

moved out `c.addOnlineStateHandler(syncTreeFunc)` from the `else` case: online handler will be refistered anyway